tsdb/db.go

```diff
@@ -595,6 +595,17 @@ func open(dir string, l log.Logger, r prometheus.Registerer, opts *Options, rngs
 		chunkPool:      chunkenc.NewPool(),
 		blocksToDelete: opts.BlocksToDelete,
 	}
+	defer func() {
+		// Close files if startup fails somewhere.
+		if err == nil {
+			return
+		}
+
+		if dbErr := db.Close(); dbErr != nil {
+			level.Error(l).Log("msg", "failed to close files after failed TSDB startup", "err", dbErr)
+		}
```

Comment:
`err` can mean anything, so we should change return variable to be clear, to maybe `returnErr`, and return close error instead of logging it. 
```suggestion
		// Close files if startup fails somewhere.
		if returnErr == nil {
			return
		}

		var merr tsdb_errors.MultiError
		merr.Add(returnErr)
		merr.Add(errors.Wrap(db.Close(), "close DB after failed startup"))
		returnErr = merr.Err()
```

Reply:
This function has a named return, so I think I should keep `err` as the variable name, right?

Changing from logging to return the closing error make a lot of sense :slightly_smiling_face:

Reply:
> This function has a named return, so I think I should keep err as the variable name, right?

We also use `err` for other errors, so the variable name could be confused